File: src/main/java/no/stunor/origo/eventorapi/model/firestore/Organisation.java
package no.stunor.origo.eventorapi.model.firestore;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.google.cloud.Timestamp;
import com.google.cloud.firestore.annotation.DocumentId;
import com.google.cloud.spring.data.firestore.Document;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Document(collectionName = "organisations")
public class Organisation {
    @JsonIgnore
    @DocumentId
    private String id;
    private String organisationId;
    private String eventorId;
    private String name;
    private String type;
    private String country;
    @JsonIgnore
    private String email;
    @JsonIgnore
    private String apiKey;
    @JsonIgnore
    private String region;
    @JsonIgnore
    private String contactPerson;
    @JsonIgnore
    private Timestamp lastUpdated;
}

File: src/main/kotlin/no/stunor/origo/eventorapi/model/organisation/Organisation.kt
package no.stunor.origo.eventorapi.model.organisation

import com.fasterxml.jackson.annotation.JsonIgnore
import no.stunor.origo.eventorapi.model.Region
import java.util.*

data class Organisation (
    var id: UUID? = null,
    @JsonIgnore var eventorId: String = "",
    var eventorRef: String = "",
    var name: String = "",
    var type: OrganisationType = OrganisationType.Club,
    var country: String = "",
    var region: Region? = null
)

enum class OrganisationType {
    Club, Region, Federation, IOF
}
